tensorflow/python/keras/callbacks.py

```diff
@@ -1370,7 +1370,13 @@ def on_epoch_end(self, epoch, logs=None):
     send = {}
     send['epoch'] = epoch
     for k, v in logs.items():
-      send[k] = v
+      # np.ndarray and np.generic are not scalar types
+      # therefore we must unwrap their scalar values and
+      # pass to the json-serializable dict 'send'
+      if isinstance(v, (np.ndarray, np.generic)):
+        send[k] = v.item()
```

Comment:
Do we know for sure that we'll get only scalar-shaped `numpy.ndarray`s? I understand that in many cases, this is just a numpy scalar. But I don't know if it's possible that the value can be a non-scalar shape (e.g., `(1, 2)`) now or in the future. 

cc @karmel @fchollet 

There are two options:
1. You may want to add a check of `len(v.shape) == 0` above. 
2. Call `x.tolist()` instead.

I think 2 is the better option, as it works for scalar and non-scalar shapes alike.

Reply:
Followed up in latter conversations.